atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/internalschema/persistence/VersionedInternalSchemaMetadataTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.internalschema.persistence;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.junit.Assume;
+import org.junit.Test;
+
+public class VersionedInternalSchemaMetadataTest {
+    private static final Path BASE_METADATA_PATH = getResourcePath("metadata-v1-base.json");
+    private static final byte[] BASE_METADATA_BYTES = readAllBytesUnchecked(BASE_METADATA_PATH);
+    private static final VersionedInternalSchemaMetadata V1_BASE = createVersionedMetadata(1, BASE_METADATA_BYTES);
+
+    @Test
+    public void metadataSemanticallyEquivalentToItselfIfCanBeDeserializedAndSemanticsAgree() {
+        Assume.assumeTrue(V1_BASE.underlyingData().isPresent());
+        assertThat(V1_BASE.knowablySemanticallyEquivalent(V1_BASE)).isTrue();
```

Comment:
why do we want to use `Assume` here? Can it cause silent broken tests at some point? I feel like it is mostly meant to be used for `Theories`.

Reply:
I originally wrote this this way because I didn't want this test to fail if we suddenly only became to deserialize v2 onwards, but we wouldn't have a test for this case if we did that which is bad. I've removed the Assume